### ISSUE TITLE
[LINUX] Editor crash when creating a new 3D project, or when trying to open some example projects

### ISSUE THREAD

**Issue by guzzard** [AtomicGameEngine/AtomicGameEngine#710]
**Creating a new _3D_ project:**
Info,  T0: Leaving post processing pipeline
[Fri Mar 18 11:21:50 2016] DEBUG: Created directory /home/username/AtomicProjects/MyGame/Resources/Models/Materials
Skipping save of existing material DefaultMaterial
[Fri Mar 18 11:21:50 2016] DEBUG: Loading resource .mdl
[Fri Mar 18 11:21:50 2016] ERROR: Could not open file /home/username/AtomicProjects/MyGame/Cache/
Could not open output file /home/username/AtomicProjects/MyGame/Cache/

**When opening some example projects (for example Basic3D or ToonTown), same error:**
Info,  T0: Leaving post processing pipeline
[Fri Mar 18 11:27:04 2016] DEBUG: Created directory /home/username/temp/AtomicExamples/Basic3D/Resources/Models/Materials
Skipping save of existing material DefaultMaterial
[Fri Mar 18 11:27:04 2016] DEBUG: Loading resource .mdl
[Fri Mar 18 11:27:04 2016] DEBUG: Loading resource Models/Materials/DefaultMaterial.material
[Fri Mar 18 11:27:04 2016] DEBUG: Loading resource Techniques/Diff.xml
[Fri Mar 18 11:27:04 2016] DEBUG: Loading resource Textures/chest.png
[Fri Mar 18 11:27:04 2016] ERROR: Could not open file /home/username/temp/AtomicExamples/Basic3D/Cache/
Could not open output file /home/username/temp/AtomicExamples/Basic3D/Cache/

Opening Basic2D and BunnyMark example projects worked fine, but they have another issue described in #713.

I'm running Arch Linux and got no errors during the build (built with ./Build_AtomicEditor.sh, build with QtCreator fails). I copied the files mentioned in #708 and the editor started fine after that, however it's not usable due to the issues described above.

**Comment by AlexandrDudnyk** [comment]
I have same problem too. Linux Mint 17.3 64 bit
`[Fri Mar 18 17:23:56 2016] DEBUG: Created directory /home/alexandr/AtomicProjects/MyGame3/Resources/Models/Materials
Skipping save of existing material DefaultMaterial

[Fri Mar 18 17:23:56 2016] DEBUG: Loading resource .mdl
[Fri Mar 18 17:23:56 2016] ERROR: Could not open file /home/alexandr/AtomicProjects/MyGame3/Cache/
Could not open output file /home/alexandr/AtomicProjects/MyGame3/Cache/
`

**Comment by JoshEngebretson** [comment]
On the first issue, examples and new project templates aren't included with a local source build,  
please try cloning and opening them from https://github.com/AtomicGameEngine/AtomicExamples

The 2nd issue needs some looking into...

**Comment by guzzard** [comment]
Hi Josh,

The projects I tried to open was cloned from the github repository you mentioned. Is there a way to turn on more verbose logging?

Also, I will create another issue with the second problem to keep the issues separate.

Kind regards
Martin

**Comment by beniwtv** [comment]
Same here... cloning the example you still get the crash.

**Comment by JoshEngebretson** [comment]
Hey guys, as you can tell the Linux build needs attention.  We have been keeping it compiling on CI, though haven't had anyone really using it on Linux.  I really want to see it get better, and quickly, though am stretched at the moment.

The CEF3 issues are kind of a show stopper for opening scripts in the editor right now.  That is a known issue.  I am not sure what is causing the project loading issues on the platform, my guess is that this is something probably trivial.  There are some reports of editor play mode issues as well, so that would also be good to address.

I am really happy to have so much interest in a solid Linux build.  I don't think it is super far off, just needs some digging into... :)

**Comment by JoshEngebretson** [comment]
@guzzard You should be able to set the log level from the command line:

```
-log <level> Change the log level, valid 'level' values: 'debug', 'info', 'warning', 'error'
```

**Comment by guzzard** [comment]
Tried to find out more with gdb. Built with "set(CMAKE_BUILD_TYPE Debug)".

Starting program: /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/AtomicEditor
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[Sat Mar 19 12:36:21 2016] INFO: Opened log file /home/username/.local/share/AtomicEditor/Logs/AtomicEditor.log
[New Thread 0x7fffe47b1700 (LWP 11696)]
[New Thread 0x7fffe3fb0700 (LWP 11697)]
[New Thread 0x7fffe37af700 (LWP 11698)]
[Sat Mar 19 12:36:21 2016] INFO: Created 3 worker threads
[Sat Mar 19 12:36:21 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/CoreData/
[Sat Mar 19 12:36:21 2016] INFO: Added resource path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/EditorData/
[Sat Mar 19 12:36:21 2016] DEBUG: Skipped autoload path 'Autoload' as it does not exist, check the documentation on how to set the 'resource prefix path'
[Sat Mar 19 12:36:21 2016] INFO: Set screen mode 2360x1475 windowed resizable
[Sat Mar 19 12:36:21 2016] INFO: Initialized input
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource Textures/Ramp.png
[Sat Mar 19 12:36:21 2016] DEBUG: Loading temporary resource Textures/Ramp.xml
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource Textures/Spot.png
[Sat Mar 19 12:36:21 2016] DEBUG: Loading temporary resource Textures/Spot.xml
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource Techniques/NoTexture.xml
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource RenderPaths/Forward.xml
[Sat Mar 19 12:36:21 2016] INFO: Initialized renderer
[New Thread 0x7fffdaede700 (LWP 11699)]
[New Thread 0x7fffd66dc700 (LWP 11700)]
[Sat Mar 19 12:36:21 2016] INFO: Set audio mode 44100 Hz stereo interpolated
[Sat Mar 19 12:36:21 2016] INFO: Initialized engine
[0319/123621:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/123621:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/123621:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/123621:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/123621:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/123621:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[0319/123621:ERROR:browser_main_loop.cc(203)] Running without the SUID sandbox! See https://code.google.com/p/chromium/wiki/LinuxSUIDSandboxDevelopment for more information on deve
loping with the sandbox on.
[New Thread 0x7fffd50fc700 (LWP 11701)]
[New Thread 0x7fffd48fb700 (LWP 11703)]
[New Thread 0x7fffd40fa700 (LWP 11704)]
[New Thread 0x7fffd38f9700 (LWP 11705)]
[New Thread 0x7fffd30f8700 (LWP 11706)]
[New Thread 0x7fffd28f7700 (LWP 11707)]
[New Thread 0x7fffd20f6700 (LWP 11708)]
[New Thread 0x7fffd18f5700 (LWP 11709)]
[New Thread 0x7fffd10f4700 (LWP 11710)]
[New Thread 0x7fffd08f3700 (LWP 11711)]
[New Thread 0x7fffd00f2700 (LWP 11712)]
[New Thread 0x7fffcf8f1700 (LWP 11713)]
[New Thread 0x7fffcf0f0700 (LWP 11714)]
[New Thread 0x7fffce8ef700 (LWP 11715)]
[New Thread 0x7fffce0ee700 (LWP 11717)]
[New Thread 0x7fffcd8ed700 (LWP 11718)]
[0319/123621:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/123621:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/123621:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/123621:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/123621:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/123621:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:WARNING:resource_bundle.cc(503)] locale resources are not loaded
[0319/123621:ERROR:extension_system.cc(71)] Failed to parse extension manifest.
[0319/123621:WARNING:resource_bundle.cc(305)] locale_file_path.empty() for locale
[0319/123621:ERROR:main_delegate.cc(722)] Could not load locale pak for en-US
[0319/123621:ERROR:main_delegate.cc(729)] Could not load cef.pak
[0319/123621:ERROR:main_delegate.cc(746)] Could not load cef_100_percent.pak
[0319/123621:ERROR:main_delegate.cc(755)] Could not load cef_200_percent.pak
[0319/123621:ERROR:main_delegate.cc(764)] Could not load cef_extensions.pak
[New Thread 0x7fffc50ca700 (LWP 11719)]
[New Thread 0x7fffc46bc700 (LWP 11721)]
Initiating Turbo Badger - version 0.1.1
[Sat Mar 19 12:36:21 2016] INFO: Initialized system user interface
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource UI/DefaultStyle.xml
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource Textures/UI.png
[New Thread 0x7fffc3ebb700 (LWP 11722)]
[New Thread 0x7fffc36ba700 (LWP 11723)]
[Sat Mar 19 12:36:21 2016] DEBUG: Loading temporary resource Textures/UI.xml
[Sat Mar 19 12:36:21 2016] DEBUG: Loading resource Fonts/Anonymous Pro.ttf
[Sat Mar 19 12:36:21 2016] DEBUG: Font face Anonymous Pro (11pt) has 624 glyphs
Skin loaded using 3 bitmaps.
[New Thread 0x7fffc206b700 (LWP 11728)]
[Sat Mar 19 12:36:22 2016] DEBUG: Started watching path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/CoreData/
[New Thread 0x7fffc186a700 (LWP 11729)]
[Sat Mar 19 12:36:22 2016] DEBUG: Started watching path /home/username/temp/AtomicGameEngine/Artifacts/AtomicEditor/Resources/EditorData/
[Sat Mar 19 12:36:22 2016] DEBUG: Window was moved to 0,0
[Sat Mar 19 12:36:22 2016] DEBUG: Window was resized to 2560x1575
[Sat Mar 19 12:36:22 2016] DEBUG: Reloading shaders
[Sat Mar 19 12:36:22 2016] DEBUG: Loading resource Shaders/GLSL/Basic.glsl
[Sat Mar 19 12:36:22 2016] DEBUG: Compiled vertex shader Basic(DIFFMAP VERTEXCOLOR)
[Sat Mar 19 12:36:22 2016] DEBUG: Compiled pixel shader Basic(DIFFMAP VERTEXCOLOR)
[Sat Mar 19 12:36:22 2016] DEBUG: Linked vertex shader Basic(DIFFMAP VERTEXCOLOR) and pixel shader Basic(DIFFMAP VERTEXCOLOR)
[New Thread 0x7fffbf19a700 (LWP 11731)]
[New Thread 0x7fffbe999700 (LWP 11732)]
[New Thread 0x7fffbe198700 (LWP 11733)]
[New Thread 0x7fffbd997700 (LWP 11734)]
[New Thread 0x7fffbd196700 (LWP 11735)]
[New Thread 0x7fffbc995700 (LWP 11736)]
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
[New Thread 0x7fffbb7b5700 (LWP 11738)]
[New Thread 0x7fffbafb4700 (LWP 11739)]
[New Thread 0x7fffba7b3700 (LWP 11740)]
[New Thread 0x7fffb9fb2700 (LWP 11741)]
[Thread 0x7fffb9fb2700 (LWP 11741) exited]
[Thread 0x7fffbd997700 (LWP 11734) exited]
[Thread 0x7fffbb7b5700 (LWP 11738) exited]
[Thread 0x7fffbe999700 (LWP 11732) exited]
[Thread 0x7fffbc995700 (LWP 11736) exited]
[Thread 0x7fffbf19a700 (LWP 11731) exited]
[Thread 0x7fffbe198700 (LWP 11733) exited]
[Thread 0x7fffba7b3700 (LWP 11740) exited]
[Thread 0x7fffbafb4700 (LWP 11739) exited]
[New Thread 0x7fffbafb4700 (LWP 11747)]
[Thread 0x7fffbd196700 (LWP 11735) exited]
[New Thread 0x7fffbd196700 (LWP 11750)]
[Thread 0x7fffbd196700 (LWP 11750) exited]
[New Thread 0x7fffbd196700 (LWP 11751)]
[Thread 0x7fffbafb4700 (LWP 11747) exited]
[New Thread 0x7fffbafb4700 (LWP 11756)]
[Thread 0x7fffbd196700 (LWP 11751) exited]
[New Thread 0x7fffbd196700 (LWP 11757)]
[Sat Mar 19 12:36:35 2016] DEBUG: Started watching path /home/username/temp/AtomicExamples/ToonTown/Resources/
[Sat Mar 19 12:36:35 2016] INFO: Added resource path /home/username/temp/AtomicExamples/ToonTown/Resources/
[Sat Mar 19 12:36:35 2016] ERROR: Could not open file /home/username/temp/AtomicExamples/ToonTown/BuildSettings.json
[Sat Mar 19 12:36:35 2016] ERROR: Could not open file /home/username/temp/AtomicExamples/ToonTown/UserPrefs.json
[New Thread 0x7fffba7b3700 (LWP 11758)]
[Sat Mar 19 12:36:35 2016] DEBUG: Started watching path /home/username/temp/AtomicExamples/ToonTown/Cache/
[Sat Mar 19 12:36:35 2016] INFO: Added resource path /home/username/temp/AtomicExamples/ToonTown/Cache/
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Models
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Scenes
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Scenes/ToonTown.scene
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Music
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Music/StoryTime.ogg
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Materials
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/UI
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Scripts
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Components
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Modules
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Textures
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Roboman
[Sat Mar 19 12:36:35 2016] INFO: CheckCacheFile:false - /home/username/temp/AtomicExamples/ToonTown/Resources/Roboman/Materials
[Sat Mar 19 12:36:35 2016] ERROR: Could not parse JSON data from /home/username/temp/AtomicExamples/ToonTown/Resources/Roboman/Robo_01.fbx.asset
AtomicEditor: /home/username/temp/AtomicGameEngine/**Source/ToolCore/Assets/AssetDatabase.cpp:353: void ToolCore::AssetDatabase::Scan(): Assertion `root.Get("version").GetInt() == ASSET_VERSION' failed.**

Thread 1 "AtomicEditor" received signal SIGABRT, Aborted.
0x00007fffef3cf2a8 in raise () from /usr/lib/libc.so.6
(gdb) bt full
#0  0x00007fffef3cf2a8 in raise () from /usr/lib/libc.so.6
No symbol table info available.
#1  0x00007fffef3d072a in abort () from /usr/lib/libc.so.6
No symbol table info available.
#2  0x00007fffef3c81b7 in __assert_fail_base () from /usr/lib/libc.so.6
No symbol table info available.
#3  0x00007fffef3c8262 in __assert_fail () from /usr/lib/libc.so.6
No symbol table info available.
#4  0x000000000117f6f5 in ?? ()
No symbol table info available.
#5  0x00007fffffffa060 in ?? ()
No symbol table info available.
#6  0x00002e8c0dafddc0 in ?? ()
No symbol table info available.
#7  0x0000000000000000 in ?? ()
No symbol table info available.

**Comment by guzzard** [comment]
output from valgrind with some more info

AtomicEditor: /home/username/temp/AtomicGameEngine/Source/ToolCore/Assets/AssetDatabase.cpp:353: void ToolCore::AssetDatabase::Scan(): Assertion `root.Get("version").GetInt() == ASSET_VERSION' failed.
==29471==
==29471== Process terminating with default action of signal 6 (SIGABRT): dumping core
==29471==    at 0xD5052A8: raise (in /usr/lib/libc-2.23.so)
==29471==    by 0xD506729: abort (in /usr/lib/libc-2.23.so)
==29471==    by 0xD4FE1B6: __assert_fail_base (in /usr/lib/libc-2.23.so)
==29471==    by 0xD4FE261: __assert_fail (in /usr/lib/libc-2.23.so)
==29471==    by 0x117F6F4: ToolCore::AssetDatabase::Scan() (AssetDatabase.cpp:353)
==29471==    by 0x117FECF: ToolCore::AssetDatabase::HandleProjectLoaded(Atomic::StringHash, Atomic::HashMap<Atomic::StringHash, Atomic::Variant>&) (AssetDatabase.cpp:445)
==29471==    by 0x1183333: Atomic::EventHandlerImpl<ToolCore::AssetDatabase>::Invoke(Atomic::HashMap<Atomic::StringHash, Atomic::Variant>&) (Object.h:281)
==29471==    by 0x12DC0E2: Atomic::Object::OnEvent(Atomic::Object*, Atomic::StringHash, Atomic::HashMap<Atomic::StringHash, Atomic::Variant>&) (Object.cpp:81)
==29471==    by 0x12DCC82: Atomic::Object::SendEvent(Atomic::StringHash, Atomic::HashMap<Atomic::StringHash, Atomic::Variant>&) (Object.cpp:287)
==29471==    by 0x116CA54: ToolCore::Project::Load(Atomic::String const&) (Project.cpp:158)
==29471==    by 0x116A765: ToolCore::ToolSystem::LoadProject(Atomic::String const&) (ToolSystem.cpp:113)
==29471==    by 0x12960C4: Atomic::jsb_class_ToolSystem_LoadProject(duk_hthread*) (JSModuleToolCore.cpp:179)
==29471==
==29471== HEAP SUMMARY:
==29471==     in use at exit: 192,536 bytes in 3,456 blocks
==29471==   total heap usage: 22,188 allocs, 18,732 frees, 1,373,368 bytes allocated
==29471==
==29471== LEAK SUMMARY:
==29471==    definitely lost: 1,336 bytes in 12 blocks
==29471==    indirectly lost: 0 bytes in 0 blocks
==29471==      possibly lost: 0 bytes in 0 blocks
==29471==    still reachable: 191,200 bytes in 3,444 blocks
==29471==         suppressed: 0 bytes in 0 blocks
==29471== Rerun with --leak-check=full to see details of leaked memory
==29471==
==29471== For counts of detected and suppressed errors, rerun with: -v
==29471== Use --track-origins=yes to see where uninitialised values come from
==29471== ERROR SUMMARY: 11620 errors from 1000 contexts (suppressed: 0 from 0)
Killed

**Comment by guzzard** [comment]
Think I found the source of this issue.

When env variable LC_NUMERIC is set to a locale which uses comma instead of punctuation the scale value of the .asset files are changed when project is opened and Cache folder is generated.

Example:
Run the editor when env variable LC_NUMERIC=sv_SE.UTF-8.
Open Basic3D example project (freshly cloned git repo).

AtomicExamples/Basic3D/Resources/Models/Chest.fbx.asset:
`{
	"version": 1,
	"guid": "b00f01294b1cc469d4db9f862c9296b7",
	"ModelImporter": {
		"scale": 0.01,
		"importAnimations": false,
		"animInfo": []
	}
}`

Is changed into (note the 0,01 after scale..):
`{
	"version": 1,
	"guid": "b00f01294b1cc469d4db9f862c9296b7",
	"ModelImporter": {
		"scale": 0,01,
		"importAnimations": false,
		"animInfo": []
	}
}`


When the editor tries to parse the file while generating the Cache folder it will crash with this error:
`AtomicGameEngine/Source/ToolCore/Assets/AssetDatabase.cpp:353: void ToolCore::AssetDatabase::Scan(): Assertion 'root.Get("version").GetInt() == ASSET_VERSION' failed.`

With a fresh clone of the example repo, if I first `export LC_NUMERIC=en_US.UTF-8` or `export LC_NUMERIC=C` and then run the editor and open up the Basic3D example project it will work fine and "0.01" is not changed into "0,01".

I have tried opening many of the example projects and I have not had any problems opening them if setting either `export LC_NUMERIC=en_US.UTF-8` or `export LC_NUMERIC=C` before starting the editor.

**Comment by guzzard** [comment]
Related rapidjson info:
Issue: https://github.com/miloyip/rapidjson/issues/72
PR: https://github.com/miloyip/rapidjson/pull/97

So seems solution would be to update rapidjson to a newer/newest version.

**Comment by zryan3** [comment]
I am testing AGE on Linux Mint (17.3) and I followed the build tutorial for QtCreator using the wiki. The editor runs at first but I get the following errors when trying to load the `Basic3D` example from AtomicExamples:

```c
[Mon Jul 11 16:37:20 2016] ERROR: Could not open file 
AtomicExamples/Basic3D/BuildSettings.json
[Mon Jul 11 16:37:20 2016] ERROR: Could not open file 
AtomicExamples/Basic3D/UserPrefs.json
```
... and
```c
[Mon Jul 11 16:37:40 2016] ERROR: Configuration::ApplyConfig - Applying a config that has not yet been populated
[0711/163740:ERROR:browser_main_loop.cc(219)] Running without the SUID sandbox! See https://chromium.googlesource.com/chromium/src/+/master/docs/linux_suid_sandbox_development.md for more information on developing with the sandbox on.
[0711/163740:ERROR:socket_posix.cc(124)] bind() returned an error, errno=98: Address already in use
[0711/163740:ERROR:devtools_http_handler.cc(229)] Cannot start http server for devtools. Stop devtools.
[0711/163741:WARNING:ipc_message_attachment_set.cc(57)] MessageAttachmentSet destroyed with unconsumed descriptors: 0/1
[Mon Jul 11 16:37:41 2016] ERROR: Opened IPC transport fd = 94
The program has unexpectedly finished.
build-AtomicGameEngine-Desktop_Qt_5_7_0_GCC_64bit-Default/Source/AtomicEditor/AtomicEditor crashed.
```

**Comment by EternalXY** [comment]
I can confirm that the issue disappears with a newer rapidjson version (tested with 1.0.2). The decimal separator for JSON data in this version is always "." regardless of locale settings.
Unfortunately the API for the newer version is slightly different, so it's not a simple drop-in replacement.

**Comment by larscwallin** [comment]
+1 for guzzards locale solution :) 

"I have tried opening many of the example projects and I have not had any problems opening them if setting either export LC_NUMERIC=en_US.UTF-8 or export LC_NUMERIC=C before starting the editor."

**Comment by JoshEngebretson** [comment]
Great,  I wonder if it would be relatively painless to cherry pick in the LC_NUMERIC fix from rapidjson

**Comment by larscwallin** [comment]
rapidjson has pretty nice unicode features it seems. This might help:

"[AutoUTF] chooses which encoding to be used according to the input or output stream. Currently, it should be used with EncodedInputStream and EncodedOutputStream."

**Comment by larscwallin** [comment]
Just realized that char table stuff won't affect locale settings  so ignore the previous post I guess :p

**Comment by guzzard** [comment]
Still doesn't work for me. Have to use LC_NUMERIC=C or projects are corrupted.. Example toon town project:
![image](https://cloud.githubusercontent.com/assets/9007314/23880612/05a1ac84-0855-11e7-98ef-58208e5e250d.png)

**Comment by JoshEngebretson** [comment]
@guzzard: Does this branch fix it? https://github.com/AtomicGameEngine/AtomicGameEngine/pull/1470

**Comment by ombre5733** [comment]
On my system, ToonTown does not work with the current master but #1470 fixes it.

**Comment by JoshEngebretson** [comment]
@ombre5733 Thanks that's great, I'll get the macOS/iOS platforms compiling in #1470 today

**Comment by JoshEngebretson** [comment]
@guzzard Ok, master should now work for you, please let me know, thanks

**Comment by guzzard** [comment]
@JoshEngebretson now the scene works as expected in the editor, but when playing the screen is just blank dark blue. When using LC_NUMERIC=C both scene and playing works as expected, so there seem to be some issue still related to the locale.

**Comment by ombre5733** [comment]
I can confirm this issue. When writing the XML file, sprintf() uses a comma as separator but reading back expects a dot. I have fixed this in PR #1477.

**Comment by JoshEngebretson** [comment]
@ombre5733 Thanks, I added a comment in #1477, I'd really like to avoid needing to modify the basic C string method usage, maybe there is a way of setting ``` LC_NUMERIC=C``` for the process or some other call to set  the locale at startup?  Of note, iOS/Android don't have locale support for C string calls

**Comment by JoshEngebretson** [comment]
@guzzard would you mind testing master now?  It looks like @ombre5733 tracked the issue down to CEF initialization :)

**Comment by guzzard** [comment]
@JoshEngebretson @ombre5733 It works for me now. Great work! :+1: :smiley:

**Comment by guzzard** [comment]
@JoshEngebretson @ombre5733 sorry have to take that back.. :cry:

Did a fresh build today, and tested the Basic 3D example C# version. When pressing play the screen is just black. But using LC_NUMERIC=C everything works fine.. the JavaScript version of Basic 3D works fine without LC_NUMERIC=C. So it seems to be related to C# in some way. However, the Racer2D example C# version works fine without LC_NUMERIC=C so not sure whats going on..

**Comment by ombre5733** [comment]
@guzzard I can confirm this. Debugging...

**Comment by JoshEngebretson** [comment]
@guzzard Racer2D constructs a scene in C# code I believe, whereas the Basic 3D project loads an XML scene.

**Comment by ombre5733** [comment]
@guzzard @JoshEngebretson Yesterday I found out that all the C# code runs in the user's locale. I have tried different things to change the locale--or culture as it is called in C#, but did not succeed. 

I have created #1497 and will fill it with some more information.
